app/src/amazonWebview/java/org/mozilla/focus/webview/FirefoxAmazonWebView.kt

```diff
@@ -112,6 +112,7 @@ internal class FirefoxAmazonWebView(
         // called by webview when clicking on a link, and not when opening a new page for the
         // first time using loadUrl().
         if (!client.shouldOverrideUrlLoading(this, url)) {
+            callback?.shouldInterceptRequest(url)
```

Comment:
This gets called twice: here and in shouldInterceptRequest. I don't think this call is necessary but if you don't, the screen flashes when `loadUrl` is initially called: I think this is because `shouldInterceptRequest` gets called from a background thread so the webView has time to blank (white) before it displays the overlay while calling `shouldInterceptRequest` from `loadUrl` is on the UI thread and happens ASAP. One hacky way to fix this is to set the default color of the WebView.

I think we should try to remove the redundant call but I'm not sure the best way to fix the flash-in issue.

Reply:
Hmm, okay - I'll file a follow-up for this, but take it out for now.